src/java/org/apache/cassandra/schema/LegacySchemaTables.java

```diff
@@ -309,12 +317,7 @@ private static List<Row> getSchemaPartitionsForTable(String schemaTableName)
 
     public static Collection<Mutation> convertSchemaToMutations()
     {
-        Map<DecoratedKey, Mutation> mutationMap = new HashMap<>();
-
-        for (String table : ALL)
-            convertSchemaToMutations(mutationMap, table);
-
-        return mutationMap.values();
+        return mutations.getUnchecked(Schema.instance.getVersion());
     }
```

Comment:
do we care about synchronizing the cache if the schema is changing underneath us?

what I think would happen is:
* working with schema_0 
* schema changes, schema uuid not yet updated
* convertSchemaToMutations invoked
* cache schema_0 uuid -> schema_1 mutations and return (matches previous behavior)
* we would continue returning schema_1 until the uuid gets updated
* after which we exceed max cache size, expiring the last value and leaving schema_1 uuid -> schema_1 mutations

Reply:
also if not too annoying to set up, a unit test to confirm the cache performs as expected (like immediately dropping the last schema version's mutations b/c max size is 1) would be nice

Reply:
Yeah, so the behavioral difference here is that until `Schema.instance.getVersion()` returns a new version, we'll return the same mutations we initially calculated until we've updated our UUID. It will eventually converge on the right mapping, as we'll never return an older schema for a newer UUID. Using your example:

- working with schema_0
- schema changes, schema uuid not yet updated
- convertSchemaToMutations invoked
- cache schema_0 uuid -> schema_1 mutations and return (matches previous behavior)
- schema changes again to schema_2 
- convertSchemaToMutations now returns schema_1 when previously it would have returned schema_2 (doesn't match previous behavior)

I may misunderstand how the schema versions get propagated in the first place, but if gossip is triggered by telling other nodes that our UUID is different, we wouldn't expect this to happen during a single schema change, only during consistent schema changes (e.g. node A is responding to a schema pull from node B associated with update to UUID 1, while node A is also processing an update to make it's own local schema UUID 2). There do seem to be some facilities for making schema changes with a "local" announce, so I could be wrong about this...

Also, not as familiar with java synchronization primitives, but would this be as "simple" as making `convertSchemaToMutations` a `synchronized` method since `mergeSchema` is already synchronized?

Reply:
yes I think that should work